pkg/services/olhttp/olrequest_test.go

```diff
@@ -187,13 +187,15 @@ func TestRead(t *testing.T) {
 							return &http.Response{StatusCode: 200, Body: r}, nil
 						}
 
-						resources := queryNResources(2, 0)
-						if req.URL.Query().Get("Cursor") != "" {
-							moreResources := queryNResources(2, 2)
```

Comment:
I have NO idea why you were first returning 2 results, then on cursor advance you were returning 4 results (including the two at the beginning). This probably was done to make tests pass, as your code was definitely not advancing through the pages.